neo4j/_async/work/session.py

```diff
@@ -239,6 +245,126 @@ async def run(self, query, parameters=None, **kwargs):
 
         return self._auto_result
 
+    async def query(self, query, parameters=None, **kwargs):
+        """
+        Run a Cypher query within an managed transaction and
+        all the retries policy will be applied.
+
+        The query is sent and the result header received
+        immediately and the :class:`neo4j.QueryResult`is
+        fetched.
```

Comment:
I don't really know what the result header is supposed to mean. If I had to guess, I's say it's the first `SUCCESS` message that contains the `fields` fields. If so, that's way too technical and in detail for the end-user.
```suggestion
        The query is sent and the full result is fetched and returned as
        :class:`neo4j.QueryResult`.
```